lib/Twig/Extensions/TokenParser/Trans.php

```diff
@@ -69,6 +73,27 @@ public function getTag()
         return 'trans';
     }
 
+    /**
+     * Overrides Twig_TokenParser::setParser.
+     * Sets the parser associated with this token parser and extract
+     * configuration from the parser environment.
+     *
+     * @param Twig_Parser $parser A Twig_Parser instance
+     */
+    public function setParser(Twig_Parser $parser)
+    {
+        parent::setParser($parser);
+
+        // Hack to allow to read the private environment from the $parser.
+        // I don't get why it was decided to deprecate the "getEnvironment"
+        // from the parser, but... whatever.
+        $env = \Closure::bind(function() {return $this->env; }, $parser, get_class($parser))->call($parser);
+
+        if ($env->hasExtension('Twig_Extensions_Extension_I18n')) {
+            $this->allow_complex = $env->getExtension('Twig_Extensions_Extension_I18n')->getComplexVars();
```

Comment:
as the `Twig_Extensions_Extension_I18n` extension is the one instantiating this TokenParser, why not passing that as a constructor argument ?